packages/headless/src/features/folding/folding-slice.ts

```diff
@@ -141,4 +171,53 @@ export const foldingReducer = createReducer(
                 payload.numberOfFoldedResults ?? state.numberOfFoldedResults,
             }
       )
+      .addCase(loadAll.pending, (state, {meta}) => {
+        const collectionId = meta.arg;
+
+        return {
+          ...state,
+          collections: state.collections.map((collection) =>
+            collection.raw[state.fields.collection] === collectionId
+              ? {...collection, isLoadingMoreResults: true}
+              : collection
+          ),
+        };
+      })
+      .addCase(loadAll.rejected, (state, {payload}) => ({
+        ...state,
+        collections: state.collections.map((collection) =>
+          collection.raw[state.fields.collection] === payload?.collectionId
+            ? {...collection, isLoadingMoreResults: false}
+            : collection
+        ),
+      }))
+      .addCase(
+        loadAll.fulfilled,
```

Comment:
Not directly related, but I was thinking, are collections ever being cleared? If not, I think this is a memory leak that could slow down the browser and eventually cause the tab to crash.

We should probably clear all collections when a new search is executed.